apps/meteor/app/authentication/server/lib/restrictLoginAttempts.ts

```diff
@@ -78,9 +79,17 @@ export const isValidLoginAttemptByIp = async (ip: string): Promise<boolean> => {
 		return true;
 	}
 
+	const lastBlockAt = (await ServerEvents.findLastBlockByIp(ip as string))?.ts;
+	const failedAttemptsSinceLastBlock = await ServerEvents.countFailedAttemptsByIpSince(ip, new Date(lastBlockAt));
+
+	if (lastBlockAt && lastBlockAt < new Date() && attemptsUntilBlock && failedAttemptsSinceLastBlock < attemptsUntilBlock) {
+		return true;
+	}
+
 	const minutesUntilUnblock = settings.get('Block_Multiple_Failed_Logins_Time_To_Unblock_By_Ip_In_Minutes') as number;
-	const willBeBlockedUntil = addMinutesToADate(new Date(lastAttemptAt), minutesUntilUnblock);
+	const willBeBlockedUntil = addMinutesToADate(new Date(), minutesUntilUnblock);
 	const isValid = moment(new Date()).isSameOrAfter(willBeBlockedUntil);
+	await saveBlockedLogin(login, willBeBlockedUntil);
```

Comment:
I think this could be made into it's own function and used in `isValidAttemptByUser`, since the same code is duplicated there.